it/thrift-fullcamel/src/test/java/server/thrift/CamelNameThriftServiceTest.java

```diff
@@ -0,0 +1,135 @@
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpRequestWriter;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.thrift.ThriftSerializationFormats;
+import com.linecorp.armeria.common.util.CompletionActions;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.thrift.THttpService;
+import com.linecorp.armeria.service.test.thrift.main.SayHelloService;
+import com.linecorp.armeria.testing.junit5.common.EventLoopExtension;
+
+/**
+ * Special test for `fullcamel` option in thrift compiler.
+ */
+class CamelNameThriftServiceTest {
+
+    private static final String FOO = "foo";
+
+    @RegisterExtension
+    static final EventLoopExtension eventLoop = new EventLoopExtension();
+
+    private TMemoryInputTransport in;
+    private TMemoryBuffer out;
+
+    private CompletableFuture<HttpData> promise;
+
+    @BeforeEach
+    void before() {
+        in = new TMemoryInputTransport();
+        out = new TMemoryBuffer(128);
+
+        promise = new CompletableFuture<>();
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(SerializationFormatProvider.class)
+    void testSync_SayHelloService_sayHello(SerializationFormat defaultSerializationFormat) throws Exception {
+        final SayHelloService.Client client = new SayHelloService.Client.Factory().getClient(
+                inProto(defaultSerializationFormat), outProto(defaultSerializationFormat));
+        client.sendSayHello(FOO);
+        assertThat(out.length()).isGreaterThan(0);
+
+        final THttpService service = THttpService.of(
+                (SayHelloService.Iface) name -> "Hello, " + name + '!', defaultSerializationFormat);
+
+        invoke(service);
+
+        assertThat(client.recvSayHello()).isEqualTo("Hello, foo!");
+    }
+
+    private TProtocol inProto(SerializationFormat defaultSerializationFormat) {
+        return ThriftSerializationFormats.protocolFactory(defaultSerializationFormat).getProtocol(in);
+    }
+
+    private TProtocol outProto(SerializationFormat defaultSerializationFormat) {
+        return ThriftSerializationFormats.protocolFactory(defaultSerializationFormat).getProtocol(out);
+    }
+
+    private static class SerializationFormatProvider implements ArgumentsProvider {
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+            return ThriftSerializationFormats.values().stream().map(Arguments::of);
+        }
+    }
+
+    private void invoke(THttpService service) throws Exception {
+        invoke0(service, HttpData.wrap(out.getArray(), 0, out.length()), promise);
+
+        final HttpData res = promise.get();
+        in.reset(res.array());
+    }
+
+    private static void invoke0(THttpService service, HttpData content,
```

Comment:
This test looks a bit hacky and I guess you refer to `ThriftServiceTest`.
We have a good suite to test Thrift service now.
You might refer to this class. https://github.com/line/armeria/blob/master/thrift0.13/src/test/java/com/linecorp/armeria/client/thrift/ClientRequestContextPushedOnCallbackTest.java#L40
I'm fine with the current test just wanted to let you know. 😄